oxd-server/src/test/java/org/xdi/oxd/server/GetTokensByCodeTest.java

```diff
@@ -82,4 +84,29 @@ public static String codeRequest(ClientInterface client, String siteId, String u
 
         return client.getAuthorizationCode(Tester.getAuthorization(), params).getCode();
     }
+
+    public static GetTokensByCodeResponse tokenByInvalidCode(ClientInterface client, RegisterSiteResponse site,
+                                                             String userId, String userSecret, String nonce) {
+
+        final String state = CoreUtils.secureRandomString();
+        final String code = CoreUtils.secureRandomString();
+
+        String testOxdId = site.getOxdId();
+
+        final GetTokensByCodeParams params = new GetTokensByCodeParams();
+        params.setOxdId(testOxdId);
+        params.setCode(code);
+        params.setState(state);
+
+        GetTokensByCodeResponse resp = null;
+
+        try {
+            resp = client.getTokenByCode(Tester.getAuthorization(), params);
+            assertTrue(false);
```

Comment:
we should assert `400, "bad_request", "'code' is invalid."` which we added above.